api/v1/linstornodeconnection_webhook.go

```diff
@@ -0,0 +1,125 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+	"strconv"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var linstornodeconnectionlog = logf.Log.WithName("linstornodeconnection-resource")
+
+func (r *LinstorNodeConnection) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-piraeus-io-v1-linstornodeconnection,mutating=false,failurePolicy=fail,sideEffects=None,groups=piraeus.io,resources=linstornodeconnections,verbs=create;update,versions=v1,name=vlinstornodeconnection.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &LinstorNodeConnection{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *LinstorNodeConnection) ValidateCreate() error {
+	linstornodeconnectionlog.Info("validate create", "name", r.Name)
+
+	errs := r.validate()
+	if len(errs) != 0 {
+		return apierrors.NewInvalid(schema.GroupKind{Group: GroupVersion.Group, Kind: "LinstorNodeConnection"}, r.Name, errs)
+	}
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *LinstorNodeConnection) ValidateUpdate(old runtime.Object) error {
+	linstornodeconnectionlog.Info("validate update", "name", r.Name)
+
+	errs := r.validate()
+	if len(errs) != 0 {
+		return apierrors.NewInvalid(schema.GroupKind{Group: GroupVersion.Group, Kind: "LinstorNodeConnection"}, r.Name, errs)
+	}
+
+	return nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *LinstorNodeConnection) ValidateDelete() error {
+	linstornodeconnectionlog.Info("validate delete", "name", r.Name)
+
+	return nil
+}
+
+func (r *LinstorNodeConnection) validate() field.ErrorList {
+	errs := ValidateControllerProperties(r.Spec.Properties, field.NewPath("spec", "properties"))
+	errs = append(errs, ValidateNodeConnectionPaths(r.Spec.Paths, field.NewPath("spec", "paths"))...)
+	errs = append(errs, ValidateNodeConnectionSelectors(r.Spec.Selector, field.NewPath("spec", "selector"))...)
+
+	return errs
+}
+
+func ValidateNodeConnectionPaths(paths []LinstorNodeConnectionPath, path *field.Path) field.ErrorList {
+	var result field.ErrorList
+
+	allNames := sets.New[string]()
+
+	for i := range paths {
+		p := &paths[i]
+		if allNames.Has(p.Name) {
```

Comment:
:smile: Good question. Turns out because we marked the name attribute as 
```
// +listType=map
// +listMapKey=name
```

We automatically get a unique constraint from the API Server. So we never actually get validation from our own code, but the API Server reports an error itself.

We also have the same "bug" in the Controller Properties validation. Not sure if we even need to keep this validation then.

Reply:
Removed the duplicate checks now, as they are unnecessary.